### DISCUSSION TITLE
Are there any native tools/frameworks to take moments of the particle data in a given cell?

### DISCUSSION THREAD

**Discussion by moisesa2** [BLAST-WarpX/warpx#4019] [Data Analysis] [answered]
As of now, I've simply done them myself for a 1D simulation using a python script but I plan on extending this case into 3D and I imagine that sorting them will be more tedious.

**Comment by dpgrote** [comment, marked as answer]
WarpX has the built in diagnostic to calculate particle moments on a cell by cell basis. See `particle_fields_to_plot` in the documentation.

**Reply by moisesa2** [reply]
Wonderful. thank you!
